lib/beaker/hypervisor/vmpooler.rb

```diff
@@ -246,5 +275,91 @@ def cleanup
       @logger.notify "Spent %.2f seconds cleaning up" % (Time.now - start)
     end
 
+    def add_disk(hostname, disk_size)
+      @logger.notify "Requesting an additional disk of size #{disk_size}GB for #{hostname}"
+
+      if !size.match /[0123456789]/ || size <= '0'
+        raise NameError.new "Disk size must be an integer greater than zero!"
+      end
+
+      begin
+        uri = URI.parse(@options[:pooling_api] + '/api/v1/vm/' + hostname + '/disk/' + disk_size.to_s)
+
+        http = Net::HTTP.new(uri.host, uri.port)
+        request = Net::HTTP::Post.new(uri.request_uri)
+        request['X-AUTH-TOKEN'] = @credentials[:vmpooler_token]
+
+        response = http.request(request)
+
+        parsed = parse_response(response)
+
+        if !parsed['ok']
+          raise RuntimeError.new "Response from #{hostname} indicates disk was not added"
+        end
+
+      rescue NameError, RuntimeError, Errno::EINVAL, Errno::ECONNRESET, EOFError,
+             Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, *SSH_EXCEPTIONS => e
+        report_and_raise(@logger, e, 'Vmpooler.add_disk')
+      end
+    end
+
+    def parse_response(response)
+      begin
+        parsed_response = JSON.parse(response.body)
+      rescue JSON::ParserError => e
+        @logger.notify "Failed to parse JSON for response from '#{host['vmhostname']}'! (failed with #{e.class})"
+      end
+    end
+
+    def disk_added?(host, disk_size, index)
+      if host['disk'].nil?
+        false
+      else
+        host['disk'][index] == "+#{disk_size}gb"
+      end
+
+    end
+
+    def get_vm(hostname)
+      begin
+        uri = URI.parse(@options[:pooling_api] + '/vm/' + hostname)
+
+        http = Net::HTTP.new(uri.host, uri.port)
+        request = Net::HTTP::Get.new(uri.request_uri)
+
+        response = http.request(request)
+      rescue RuntimeError, Errno::EINVAL, Errno::ECONNRESET, EOFError,
+             Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, *SSH_EXCEPTIONS => e
+        @logger.notify "Failed to connect to vmpooler while getting VM information!"
+      end
+
+    end
+
+    def wait_for_disk(hostname, disk_size, index)
+
+      response = get_vm(hostname)
+      parsed = parse_response(response)
+
+      @logger.debug "Waiting for disk"
+
+      attempts = 0
+
+      while (!disk_added?(parsed[hostname], disk_size, index) && attempts < 20)
```

Comment:
good to know @eputnam. I think that the other method there, the [repeat_for](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/shared/repetition.rb#L5) would then be appropriate. But, it's not absolutely necessary to use that method, just wanted to make sure you knew those methods were available to you.